libs/common/src/main/java/org/opensearch/common/annotation/PublicApi.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+/**
+ * Stable public APIs that retain source and binary compatibility within a major release.
+ * These interfaces can change from one major release to another major release
+ * (e.g. from 1.0 to 2.0). The types marked with this annotations could only expose
+ * other {@link PublicApi} or {@link ExperimentalApi} types as public members.
+ *
+ * @opensearch.api
+ */
+@Documented
+@Target({ ElementType.TYPE, ElementType.PACKAGE, ElementType.METHOD, ElementType.CONSTRUCTOR })
+@PublicApi
+public @interface PublicApi {
+    /**
+     * Version when this API was released
+     */
+    String since() default "";
```

Comment:
Adding "" for now, would be difficult to reverse engineer all API, may be `1.0.0` as a placeholder would be better, @peternied ?

Reply:
Using `2.10.0` since this is the version we could get (:crossed_fingers:) these annotations in